File: targets/jsts-web/src/vulns/vuln16_verbose.ts
import { Router } from "express";
import { db } from "../store";
export const router = Router();
router.get("/vuln16/lookup", (req, res) => {
  try { db.prepare("SELECT * FROM users WHERE id = " + String(req.query.id || "")).get(); res.json({ status: "ok" }); }
  catch (e: any) {
    // VULN:VULN-16:CWE-209:logic full error/stack returned to the client
    res.status(500).send("error:\n" + (e.stack || e.message));
  }
});

File: targets/python-web/vulns/vuln16_verbose.py
import traceback
from flask import Blueprint, request
from store import get_db
bp = Blueprint("vuln16", __name__)

@bp.get("/vuln16/lookup")
def lookup():
    uid = request.args.get("id", "")
    try:
        con = get_db()
        con.execute("SELECT * FROM users WHERE id = " + uid).fetchone()
        return {"status": "ok"}
    except Exception:
        # VULN:VULN-16:CWE-209:logic full traceback returned to the client
        return "error:\n" + traceback.format_exc(), 500
